conans/test/functional/compression/compression_test.py

```diff
@@ -0,0 +1,454 @@
+                        st = os.stat(filename)
+                        if st.st_ino in inodes:
+                            ti.type = tarfile.LNKTYPE
+                            ti.linkname = inodes[st.st_ino]
+                            ti.size = 0
+                            ti.mode = st.st_mode
+                            ti.mtime = st.st_mtime
+                            f.addfile(tarinfo=ti)
+                        else:
+                            inodes[st.st_ino] = archname
+                            ti.size = st.st_size
+                            ti.mode = st.st_mode
+                            ti.mtime = st.st_mtime
+                            with open(filename, 'rb') as file_handler:
+                                f.addfile(tarinfo=ti, fileobj=file_handler)
+                for dirname in dirnames:
+                    filename = os.path.join(root, dirname)
+                    archname = os.path.relpath(filename, d)
+                    ti = tarfile.TarInfo(archname)
+                    if os.path.islink(filename):
+                        ti.type = tarfile.SYMTYPE
+                        ti.linkname = os.readlink(filename)
+                        ti.size = 0
+                        f.addfile(tarinfo=ti)
+                    else:
+                        st = os.stat(filename)
+                        ti.type = tarfile.DIRTYPE
+                        ti.size = 0
+                        ti.mode = st.st_mode
+                        f.addfile(tarinfo=ti)
+
+    def decompress(self, d):
+        with tarfile.open(name='test' + self.ext, mode=self.read_mode) as f:
+            f.extractall(path=d)
+            f.close()
+
+class TarGzCompressor(TarCompressor):
+    read_mode = "r:gz"
+    write_mode = "w:gz"
+    ext = ".tgz"
+
+
+class TarBz2Compressor(TarCompressor):
+    read_mode = "r:bz2"
+    write_mode = "w:bz2"
+    ext = ".tbz2"
+
+
+class TarLZMACompressor(TarCompressor):
+    read_mode = "r:xz"
+    write_mode = "w:xz"
+    ext = ".txz"
+
+
+compressors = [{"compressor": ZipCompressor()},
+               {"compressor": TarCompressor()},
+               {"compressor": TarGzCompressor()},
+               {"compressor": TarBz2Compressor()}]
+
+if sys.version_info.major >= 3:
+    compressors.append({"compressor": TarLZMACompressor()})
+
+
+# check if we can actually support symlinks:
+# some operation systems might not support them
+# e.g. Windows 10 requires either privileges or developer mode
+# also only certain file systems have support (e.g. NTFS does, but FAT32 doesn't)
+def symlinks_supported():
+    if not hasattr(os, "symlink"):
+        return False
+    client = TestClient()
+    try:
+        client.save({"a": ""})
+        with chdir(client.current_folder):
+            os.symlink("a", "b")
+            return os.path.islink("b")
+    except OSError:
+        return False
+
+
+def hardlinks_supported():
+    if not hasattr(os, "link"):
+        return False
+    client = TestClient()
```

Comment:
Same, no need to instance a TestClient()

Reply:
removed TestClient